packages/react-mui-store-file-viewer/src/components/StoreDownloadDataGrid/index.tsx

```diff
@@ -181,7 +190,21 @@ const StoreDownloadDataGrid: FC<StoreDataGridProps> = ({
     );
   }, [defaultVisibilityModel, columns, isFiltered]);
 
+  useEffect(() => {
+    // For showing a single shared folder as the root,
+    // auto-select top-level folder if virtualRoot flag is set
+    if (virtualRoot && data.count === 1) {
+      const obj = data.objects[0];
+      if (obj.type === "folder" && obj.folder === "/") {
+        setFolder("/" + data.objects[0].name);
+      }
```

Comment:
should this if statement block also setFirstLoad to false, otherwise would it keep opening folders?

Reply:
Calling setFirstLoad(false) in this if statement causes a flash of the top folder, which I want to prevent. The `else if` part handles setting firstLoad in any other case. The `data.last` parameter will only be an empty string on init. After any list call, it will be undefined if no results, or have a value.

Reply:
but what if data.count stays at 1? Would switching `virtualRoot && data.count === 1` to `virtualRoot && data.count === 1 && firstLoad` fix the flash?

Reply:
Good point on data.count staying 1. Added an else that sets firstLoad to false. Tested and it handles that case.